gcp_variant_transforms/libs/beam_util.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Wrapper for Beam classes, e.g., Metrics.
+
+This is used to avoid direct dependency on Beam in our library methods/classes
+such that they are more modular and easier to test. Preferably instances
+of classes in this module should be injected into library classes/methods
+instead of direct instantiation.
+"""
+
+from __future__ import absolute_import
+
+import logging
+
+from apache_beam.runners import runner  # pylint: disable=unused-import
+from apache_beam.metrics import Metrics
+from apache_beam.metrics.metric import MetricsFilter
+
+_METRICS_NAME_SPACE = 'VT_metrics_name_space'
+
+
+class BaseCounter(object):
```

Comment:
Not sure what value this base class adds in python world?

Reply:
You don't need base class for the benefits you mentioned. Define the child and test class as before just without inheritance and it works. I'll leave it to you if you prefer this way. :)

Reply:
So as discussed offline, the benefit of BaseCounter is that (i) as an interface it documents the intended API of Counter implementations (we can also impose it to some extent in Python with "types" we are adding) and (ii) Having a base implementation let us implement the Null object pattern easily.